Mathlib/Analysis/SpecialFunctions/Pow/Complex.lean

```diff
@@ -112,53 +112,40 @@ theorem cpow_sub {x : ℂ} (y z : ℂ) (hx : x ≠ 0) : x ^ (y - z) = x ^ y / x
 theorem cpow_neg_one (x : ℂ) : x ^ (-1 : ℂ) = x⁻¹ := by simpa using cpow_neg x 1
 #align complex.cpow_neg_one Complex.cpow_neg_one
 
+lemma cpow_int_mul (x : ℂ) (n : ℤ) (y : ℂ) : x ^ (n * y) = (x ^ y) ^ n := by
```

Comment:
This statement looks backwards to me, I'd expect
```suggestion
lemma cpow_int_mul (x : ℂ) (n : ℤ) (y : ℂ) : x ^ (n * y) = (x ^ n) ^ y := by
```

Reply:
But this isn't true for `x = -I`, `n = 2`, `y = 1/2`.

Reply:
In that case, I'd be inclined to prove `cpow_mul_int` first, since that presumably saves you commutativity in the proof.
A comment pointing out that trap feel like it might be a good iea

Reply:
Do we have a lemma for the case when `(x ^ y) ^ n = x ^ (y * n)` _is_ true? Or is it awkward to state?

Reply:
The condition isn't too awkward to state:
```lean
import Mathlib.Analysis.SpecialFunctions.Complex.Log
import Mathlib.Analysis.SpecialFunctions.Pow.Complex

example (x y : ℂ) (n : ℤ) : -Real.pi < x.arg * n → x.arg * n ≤ Real.pi → x ^ (n * y) = (x ^ n) ^ y := by
  convert Complex.cpow_mul y <;> simp [Complex.log_im]
```

Reply:
> since that presumably saves you commutativity in the proof.

No, it doesn't.